docs/configuration.md

```diff
@@ -0,0 +1,71 @@
+# Configuring Spectral
+
+## Example config file (spectral.yaml)
+
+```yaml
+rules:
+    - rule-name:
+        given: $..parameters[*]
+        then:
+            field: description
+            function: truthy        
+```
+
+## Rules
+
+Rules are highly configurable. There are only few required parameters but the optional ones gives powerful flexibility.
+
+TBD: replace TS interfaces with a nice table.
+
+```ts
+export interface IRule<T = string, O = any> {
+  type?: RuleType;
+
+  // A short summary of the rule and its intended purpose
+  summary?: string;
+
+  // A long-form description of the rule formatted in markdown
+  description?: string;
+
+  // The severity of results this rule generates
+  severity?: DiagnosticSeverity;
+
+  // Tags attached to the rule, which can be used for organizational purposes
+  tags?: string[];
```

Comment:
I'd rather *not* make it optional. Provide `[]` if you do not want to provide any tag?